Update CAMARA-API-access-and-user-consent.md login_hint example

AxelNennker opened the pull request:
CIBA does require the authorization request to be authenticated but allows a wide range of methods.

Removing "Basic" from the example

#### What type of PR is this?
* cleanup

#### What this PR does / why we need it:

Clarification that Basic authentication is not important for the example. The login_hint part is.


#### Which issue(s) this PR fixes:

Fixes #104

jpengar reviewed (approved):
LGTM

jpengar commented on documentation/CAMARA-API-access-and-user-consent.md:
````diff
@@ -281,7 +281,7 @@ The API invoker has to provide in the authorization request (/bc_authorize) a lo
     ```
     POST /bc-authorize HTTP/1.1
-    Authorization: Basic {Credentials}
+    Authorization: ...
````
Actually, we can remove the authorization header entirely and just leave the example without explicitly describing client authentication, since this example is really meant for login_hint. I say this because the Authorization header would not apply to private_key_jwt, for example.

AxelNennker commented:
As per your comment @jpengar I removed the Authorization header and added "client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer"

@rartych please review

AxelNennker commented:
@jpengar @rartych @sebdewet please review

jpengar reviewed (approved):
> @jpengar @rartych @sebdewet please review

LGTM, but #155 removes that text and links directly to the new CAMARA-Security-Interoperability.md doc. We may consider closing this PR if this part of the document is eventually removed anyway.

> The API invoker has to provide in the authorization request (/bc_authorize) a login_hint with a valid user identifier together with the application credentials (the client_id of the final application requesting access to the data) and indicate the purpose for accessing the data (Step 1). The login_hint possible values and format for CAMARA and the way to declare a purpose when accessing the CAMARA APIs is defined in [the CAMARA Security and Interoperability Profile](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md).

AxelNennker commented:
@jpengar I find it difficult to resolve the merge conflict. I guess that some other changes to a&uc lead to them.
Could you please resolve the one merge conflict?

jpengar commented:
> @jpengar I find it difficult to resolve the merge conflict. I guess that some other changes to a&uc lead to them. Could you please resolve the one merge conflict?

#155 has been merged. So the part of the document that this PR modifies is no longer in the main branch document. As suggested in https://github.com/camaraproject/IdentityAndConsentManagement/pull/105#pullrequestreview-2082316199, I think we should just close this PR.

AxelNennker commented:
Replacing "Authorization Basic" was the point of this PR and that was done. Closing this without resolving the conflict
